### PR TITLE
enhance architecture of LowercaseKeysDict

### PR DESCRIPTION
**Pull request by buxx**

### CODE REVIEW

File: hapic/ext/aiohttp/context.py [hunk 1]
```diff
@@ -62,8 +62,8 @@ def form_parameters(self):
     @property
     def header_parameters(self):
```
Comment (lebouquetin) [inline hapic/ext/aiohttp/context.py]: missing typing

File: hapic/util.py [hunk 2]
```diff
@@ -1,53 +1,91 @@
...
     """
 
-    def get(
-        self, key: typing.Any, default_value: typing.Any = None
-    ) -> typing.Any:
+    @staticmethod
+    def check_key(key: str) -> None:
+        try:
+            if key.lower() != key:
+                raise HapicException(
```
Comment (lebouquetin) [inline hapic/util.py]: Should raise an exception inheriting from `HapicException`, like `NotLowercaseCaseException` ?

File: tests/unit/test_utils.py [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
+    def test_unit__set__error__refuse_key(self):
+        lowercase_dict = LowercaseKeysDict()
+
+        with pytest.raises(HapicException):
+            lowercase_dict.setdefault("FOO", "bar")
+
+    def test_unit__update__error__refuse_key(self):
+        lowercase_dict = LowercaseKeysDict()
+
+        with pytest.raises(HapicException):
```
Comment (lebouquetin) [inline tests/unit/test_utils.py]: Exception should be specific for a specific case, no?

File: hapic/ext/aiohttp/context.py (hunk 1 above)
Comment (buxx) [reply hapic/ext/aiohttp/context.py]: added

File: hapic/util.py (hunk 2 above)
Comment (buxx) [reply hapic/util.py]: You're right. It is done.

File: tests/unit/test_utils.py (hunk 3 above)
Comment (buxx) [reply tests/unit/test_utils.py]: Changed.

**Pull request merged**